src/stdune/path.ml

```diff
@@ -2,17 +2,39 @@ let explode_path =
   let rec loop path acc =
     let dir  = Filename.dirname  path in
     let base = Filename.basename path in
-    let acc = base :: acc in
     if dir = Filename.current_dir_name then
+      base :: acc
+    else if dir = path then
```

Comment:
I remember doing similar things in the past but I don't remember why. What case does this catch?

Reply:
It will not work for absolute paths otherwise. As `Filename.dirname "/" = "/"` it will just loop forever.

Reply:
ok